buildSrc/subprojects/plugins/src/main/kotlin/org/gradle/plugins/performance/PerformanceTestPlugin.kt

```diff
@@ -423,31 +409,29 @@ class PerformanceTestPlugin : Plugin<Project> {
             jvmArgs("-Xmx3g", "-XX:+HeapDumpOnOutOfMemoryError")
 
             dependsOn(prepareSamplesTask)
-            finalizedBy(performanceReportTask)
+
+            registerTemplateInputsToPerformanceTest()
 
             mustRunAfter(tasks.withType<ProjectGeneratorTask>())
             mustRunAfter(tasks.withType<RemoteProject>())
             mustRunAfter(tasks.withType<JavaExecProjectGeneratorTask>())
-
-            doFirst {
-                channel?.let { channel ->
-                    performanceReportTask.get().systemProperty(PropertyNames.channel, channel)
```

Comment:
Good catch, but if IUUC, you're saying I omitted `channel` system property in report generation `JavaExec` task - now we're using `PerformanceTestJvmArgumentsProvider` so that system property can't be seen by report generation task any more. I added that system property back.